importlab/resolve.py

```diff
@@ -151,11 +201,18 @@ def resolve_import(self, item):
         for fs in self.fs_path:
             for module_name, path in files:
                 f = self._find_file(fs, path)
-                if f:
-                    if item.is_relative():
-                        return Relative(f, module_name)
-                    else:
-                        return Local(f, module_name, fs)
+                if not f:
+                    continue
+                if item.is_relative():
+                    if self.current_module:
+                        module_name = get_absolute_name(
+                                self.current_module.package_name, module_name)
+                    # TODO(martindemello): If we do have a current_module,
```

Comment:
Would returning a module of the same type allow us to get rid of Relative altogether?

Reply:
i don't think so - we still need a way to pass the information that this was resolved relative to another filepath up to the calling code in the case that we cannot figure out a module name here. also it's an actual semantic difference because it has its own resolution mechanism, though i guess you could argue that we need not expose that mechanism to the user.

Reply:
it's not used within importlab, but pytype-all might need it in the event that we cannot populate module_name. but let me try reworking the code on the assumption that we will always have a module name and see where that gets us; i can always put it back later if needed.